File: ios/JJMproject/Classes/JSBridge/Modules/Notification/NotificationModule.m
//
//  NetworkModule.m
//  MarioFramework
//
//  Created by MeePwn on 2017/11/15.
//  Copyright © 2017年 Facebook. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(NotificationModule, NSObject)

RCT_EXTERN_METHOD(sendLocalNotification:(double)after alertBody:(NSString *)alertBody alertAction:(NSString *)alertAction badgeNumber:(int)badgeNumber)
RCT_EXTERN_METHOD(sendLocalNotification:(NSString *)alertBody alertAction:(NSString *)alertAction badgeNumber:(int)badgeNumber)

@end

File: ios/JJMproject/Classes/JSBridge/Modules/Notification/NotificationModule.swift
//
//  NotificationModule.swift
//  MarioFramework
//
//  Created by MeePwn on 2017/11/15.
//  Copyright © 2017年 Facebook. All rights reserved.
//

import UIKit
import Async

@objc(NotificationModule)
class NotificationModule: NSObject {

  private lazy var notificationUtility: NotificationUtility = {
    return NotificationUtility()
  }()
  
  @objc func sendLocalNotification(_ alertBody: String, alertAction: String, badgeNumber: Int) {
    Async.main {
      self.notificationUtility.sendLocalNotification(alertBody: alertBody, alertAction: alertAction, badgeNumber: badgeNumber)
    }
  }
  
  @objc func sendLocalNotification(_ after: Double, alertBody: String, alertAction: String, badgeNumber: Int) {
    Async.main {
      self.notificationUtility.sendLocalNotification(after: after, alertBody: alertBody, alertAction: alertAction, badgeNumber: badgeNumber)
    }
  }
  
}
